resources/sts/4.14/hypershift/sts_extended_hcp_support_permission_policy.json

```diff
@@ -62,49 +137,300 @@
+                },
+                "ForAllValues:StringEquals": {
+                    "aws:TagKeys": "red-hat-managed"
+                }
+            }
+        },
+        {
+            "Sid": "NetworkVerifier2",
+            "Effect": "Allow",
+            "Action": [
                 "ec2:RunInstances",
                 "ec2:DescribeInstanceTypes",
-                "ec2:CreateTags",
                 "ec2:GetConsoleOutput"
             ],
+            "Resource": "arn:aws:ec2:*:*:instance/*",
+            "Condition": {
+                "StringEquals": {
+                    "aws:ResourceTag/red-hat-managed": "true"
+                }
+            }
+        },
+        {
+            "Sid": "Ip",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeCoipPools",
+                "ec2:DescribeIpv6Pools",
+                "ec2:DescribePublicIpv4Pools",
+                "ec2:GetAssociatedIpv6PoolCidrs"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "InternetGateway",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeEgressOnlyInternetGateways",
+                "ec2:DescribeInternetGateways"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "Describetag",
+            "Effect": "Allow",
+            "Action": [
+               "ec2:DescribeTags" 
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "PrefixLists",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribePrefixLists",
+                "ec2:DescribeManagedPrefixLists"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "PrefixListsentries",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:GetManagedPrefixListEntries"
+            ],
+            "Resource": "arn:aws:ec2:*:*:prefix-list/*"
+        },
+        {
+            "Sid": "Image",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeImages"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "ImagaeAttibute",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeImageAttribute"
+            ],
+            "Resource": "arn:aws:ec2:*:*:image/*"
+        },
+        {
+            "Sid": "IdFormat",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribePrincipalIdFormat",
+                "ec2:DescribeIdentityIdFormat",
+                "ec2:DescribeIdFormat",
+                "ec2:DescribeAggregateIdFormat"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "AddressAttribute",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeAddressesAttribute"
+            ],
+            "Resource": "*"
```

Comment:
Any way these can be restricted more?

Reply:
resource set to "*" because "elastic-ip" failed with dry-run
`
aws ec2 --dry-run describe-addresses-attribute --allocation-ids eipalloc-abcdef01234567890 --attribute domain-name
An error occurred (UnauthorizedOperation) when calling the DescribeAddressesAttribute operation: You are not authorized to perform this operation. User: arn:aws:sts::301721915996:assumed-role/sdq-ci-tgsde-HCP-ROSA-Support-Role/5a9d529885b8d55fb4b1902fd87f9c2f is not authorized to perform: ec2:DescribeAddressesAttribute because no identity-based policy allows the ec2:DescribeAddressesAttribute action
—---> change resource to /*
`
No red-hat tags associated with elastic IP address